internal/tasks/manager.go

```diff
@@ -0,0 +1,87 @@
+package tasks
+
+import (
+	"context"
+	"time"
+
+	"github.com/flightctl/flightctl/internal/store"
+	"github.com/flightctl/flightctl/internal/tasks/repotester"
+	"github.com/flightctl/flightctl/pkg/thread"
+	"github.com/google/uuid"
+	"github.com/sirupsen/logrus"
+)
+
+type ResourceReference struct {
+	OrgID uuid.UUID
+	Kind  string
+	Name  string
+}
+
+type TaskManager struct {
+	log        logrus.FieldLogger
+	ctx        context.Context
+	cancelFunc context.CancelFunc
+	channels   map[string]chan ResourceReference
+	store      store.Store
+}
+
+const (
+	FleetTemplateRollout = "fleet-template-rollout"
+
+	ChannelSize = 20
```

Comment:
buffered channel is great but remember every op in a txn will be sent to the channel so this might need to be bigger. if the callback can be blocked by write to channel we are blocking the store I believe. TGhis might need to be configurable. depending on scale or a larger default.